app/client/components/LinkingState.ts

```diff
@@ -9,150 +9,220 @@ import {FilterColValues, QueryOperation} from "app/common/ActiveDocAPI";
+      const updateHolder = Holder.create(this);
+
+      // source data table could still be loading (this could happen after changing the group-by
+      // columns of a linked summary table for instance). Define an _update function to be called when data loads
+      const _update = () => {
+        if (srcSection.isDisposed() || srcSection.table().groupByColumns().length === 0) {
+          //srcSection disposed can happen transiently. Can happen when deleting tables and then undoing?
+          //Tests nbrowser/LinkingErrors and RawData might hit this case
+          //groupByColumns === [] can happen if we make a summary tab [group by nothing]. (in which case: don't filter)
+          _filterState(EmptyFilterState);
+          return;
         }
-        const srcRowId = srcSection.activeRowId();
-        for (const c of srcSection.table().groupByColumns()) {
-          const colId = c.colId();
-          const srcValue = srcTableData.getValue(srcRowId as number, colId);
-          result.filters[colId] = [srcValue];
-          result.operations[colId] = 'in';
-          if (isDirectSummary && isListType(c.summarySource().type())) {
-            // If the source groupby column is a ChoiceList or RefList, then null or '' in the summary table
-            // should match against an empty list in the source table.
-            result.operations[colId] = srcValue ? 'intersects' : 'empty';
-          }
+
+        //Make a MultiHolder to own this invocation's objects (disposes of old one)
+        //TODO (MultiHolder in a Holder is a bit of a hack, but needed to hold multiple objects I think)
+        const updateMultiHolder = MultiHolder.create(updateHolder);
+
+        //Make one filter for each groupBycolumn of srcSection
+        const resultFilters: (ko.Computed<FilterState>|undefined)[] = srcSection.table().groupByColumns().map(srcGCol =>
+          this._makeFilterObs(srcGCol, summaryGetCorrespondingCol(srcGCol, tgtSection.table()), updateMultiHolder)
+        );
+
+        //If any are undef, error out
+        if(resultFilters.some((f) => f === undefined)) {
+          console.warn("LINKINGSTATE: some of filters are undefined", resultFilters);
+          _filterState(EmptyFilterState);
+          return;
         }
-        _filterColValues(result);
-      }
-    } else if (srcSection.selectedRowsActive()) {
-      this.filterColValues = this._srcCustomFilter('id', 'in');
-    } else {
-      const srcValueFunc = srcColId ? this._makeSrcCellGetter() : identity;
-      if (srcValueFunc) {
+
+        //Merge them together in a computed
+        const resultComputed = updateMultiHolder.autoDispose(ko.computed(() => {
+          return merge({}, ...resultFilters.map(filtObs => filtObs!())) as FilterState;
+        }));
+        _filterState(resultComputed());
+        resultComputed.subscribe((val) => _filterState(val));
+      }; // End of update function
+
+      //Call when data loads, also call now to be safe
```

Comment:
The "source data table could still be loading ..." comment?
It's still there, just defined at the top of the _update function. 

I had to move the `addListener` to below the function when switching it to an arrow-func, since the function no longer gets hoisted.